include/hip/hcc_detail/hiprtc.h

```diff
@@ -84,7 +84,7 @@ hiprtcResult hiprtcGetCodeSize(hiprtcProgram prog, std::size_t* codeSizeRet);
 
 namespace hip_impl
 {
-    std::string demangle(const char* mangled_expression);
+    char* demangle(const char* mangled_expression);
 }
```

Comment:
Maybe to change the type of `mangled_expression` to `const std::string&` instead?

Reply:
I believe I had tried that and it wasnt happy. Just added you to the local thread.